addons/wiki/models.py

```diff
@@ -146,11 +179,10 @@ def save(self, *args, **kwargs):
             self.wiki_page.node.update_search()
         self.wiki_page.modified = self.created
         self.wiki_page.save()
-        self.spam_check()
+        self.check_spam()
         return rv
 
-    def spam_check(self):
-        # Since wiki_pages_current will be removed from Node model, when a new WikiVersion is saved, trigger a spam check.
+    def _get_user_and_request_headers(self):
```

Comment:
Mild recommendation: maybe you can remove this function by splitting into _get_current_user and a generic get request headers function. There's just no specific wiki behavior here I can see and returning things as tuples can get problematic.

Reply:
This method uses a pre-existing method [get_request_and_user_id](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/utils/requests.py#L40) that is also returning a tuple.  Returned a tuple here so I didn't have to make a separate call to this same function.